### PR TITLE
Update python bindings to python3

### PR DESCRIPTION
**Pull request by WSLUser**
Closes https://github.com/libsixel/libsixel/issues/33

I switched as many things as I could find to ensure python3 was explicitly used. Very little changes needed to actually get it running. If the build or CI are broke by one of the changes  for python, the commits are one file each so it's easy to fix up if needed. Let me know if what if anything needs changing. Also feel free to make edits as I'm no build expert if there is a problem. 

VSCode was giving linter warnings about things needing explicit imports but didn't mess with those to keep the changes minimal.

Also: I'd like to get a `pyproject.toml` running for the python bindings as I'd rather we switched from pip to poetry. I'd appreciate anybody who could help get that set up.

### CODE REVIEW

**Comment by dankamongmen** [comment]
nice work, thanks! i'll take a close look later today.

File: python/setup.py
Comment (dankamongmen) [inline python/setup.py:2]: is this our python module's version? ought it just match the main revision?

File: README.md
Comment (dankamongmen) [inline README.md:296]: i'd drop the 3 here

**Review by dankamongmen — CHANGES_REQUESTED** [review]

File: python/setup.py
Comment (WSLUser) [reply python/setup.py:2]: I figured it would be released with next version

File: README.md
Comment (WSLUser) [reply README.md:296]: This just reflects the change. But I would prefer the second one to just say python if I figure out which place needs to be change back.

Comment (dankamongmen) [reply README.md:296]: yeah i'd prefer it just be `--enable-python`

File: python/setup.py
Comment (dankamongmen) [reply python/setup.py:2]: go ahead and set it to the same thing now, and please add the line to `tools/release` to update it when we do a release, otherwise it'll surely go out of sync

File: NEWS
Comment (dankamongmen) [inline NEWS:2]: you'll need to get @ctrlcctrlv to agree to new maintainer

**Review by dankamongmen — CHANGES_REQUESTED** [review]

File: NEWS
Comment (WSLUser) [reply NEWS:2]: I only put this as I wasn't sure if I needed to. Could I say contributor or do I need to say anything? I wanted to ensure that the change was documented for users when the next release happens but saw the format listing and thought it was simply a way to give credit. I'm not interested in being an actual maintainer, just contribute when I can for things I feel comfortable with.  Let me know what I should put here or just revert the commit.

File: README.md
Comment (WSLUser) [reply README.md:296]: So I think this actually comes from  `setup.py` under setup at the top. I'll try giving it a quick spin using just `--enable-python`

File: python/setup.py
Comment (WSLUser) [reply python/setup.py:2]: Not seeing where I'd update `tools/release`. Doesn't seem to check the specific versions of the bindings. If we need to add that, I'd honestly prefer you did that, especially since you created the file and understand how it works better than I do. From what I can determine, it seems it shouldn't matter. Of course no idea where OLD_VERSION $1 is supposed to mean other than possibly the version of libsixel that existed prior to this fork.

Comment (WSLUser) [reply python/setup.py:2]: I did notice the published Pypi version documented is not the same version however. Might need to update the doc.

Comment (WSLUser) [reply python/setup.py:2]: Confirmed: https://pypi.org/project/libsixel-python/ points to 0.5.0. We'll need to submit a new PyPi to get the python module updated. @ctrlcctrlv would have to be one to do it though.

**Comment by WSLUser** [comment]
So I've noticed that building this is more difficult than I thought. I'm having an issue just getting meson setup build working. It's pointing to my python line as the culprit but it needs to detect python3 and thought it would detect it. It actually tries doing something if I switch the last python name but still fails. I'm not sure what Meson wants from me to detect python correctly. Were the python bindings even working before? Seems switching to Meson broke the python install.

**Comment by dankamongmen** [comment]
.nodnod, i didn't mean to make demands that would greatly complicate this work. i can probably handle adding anything i requested once we merge this.

**Comment by WSLUser** [comment]
My only issue running setup.py directly is it's asking for libsixel, which I believe is generated by meson and gets imported to create a pythonc file. I can accept it being pythonc if the c compiler is actually improving the perf but if it isn't, I'd really prefer it to be a pure python implementation. I think the only part making it pythonc is the fact it's importing the libsixel C library, at least that's been my impression.

File: NEWS
Comment (WSLUser) [reply NEWS:2]: I got a different entry here I can put so I don't need to say maintainers. I just glossed over other contributor entries as they were not where I expected them mentioned at first.

**Comment by WSLUser** [comment]
I've got some promise, I added python as a libsixel dep and got ``meson.build:91:0: ERROR: Tried to access unknown option "python3".`` I think the "import python(3)" option is off. Fix that and we'll probably have a working build again.

**Comment by WSLUser** [comment]
So I almost have it working but it sets python as a dependency but doesn't install it by default. I have a rogue line somewhere to track down getting called by message. I'm committing what I have in case I don't figure out today.

**Comment by WSLUser** [comment]
And it's now working. Now comes the scrutinizing if this is the approach we want for building it. If not, somebody else will need to update the meson. I got it working and that's enough for me. As far as I can tell, it's not installing the python bindings by default, just ensuring you have python3 installed. Only caveat is if a system doesn't have python3 installed. That magic trickery can be handled by somebody else though.

**Comment by WSLUser** [comment]
Spoke too soon. I removed errors from `meson setup build` but I'm unable to use the` --enable-python` option. This is where it might be going beyond me.

File: README.md
Comment (ctrlcctrlv) [inline README.md:296]: User software may consist of legacy Python 2 scripts. It would be better to add Python 3 support like this:

* if `--enable-python` ⇒ enable python3, give warning that this used to enable python2 and that function is now behind `--enable-python2`
* if `--enable-python2` ⇒ enable python2
* if `--enable-python3` ⇒ enable python3 (explicit version)

Comment (dankamongmen) [reply README.md:296]: i know that most distros are not allowing python2 packages, though they are generally still shipping python2 interpreters. i don't want to put any effort, going forward, into maintaining python2 code -- legacy python2 scripts ought by now have been updated to python3. python2 was EOLd over a year ago.

i'm very hesitant to continue shipping python2. if someone truly needs it, they can grab an old release, no?

wouldn't the python2 scripts need to have an explicit `/usr/bin/env python2` or whatever to work on most distros? are we proposing to rewrite the shebang lines at installation time?

**Comment by WSLUser** [comment]
If you need to make a change, edits are permitted

**Comment by dankamongmen** [comment]
i like it!

**Comment by eli-schwartz** [comment]
> Also: I'd like to get a `pyproject.toml `running for the python bindings as I'd rather we switched from pip to poetry. I'd appreciate anybody who could help get that set up.

Why poetry specifically? (Disclaimer: I love setuptools -- particularly if you use setup.cfg instead of setup.py -- and hate poetry.)

Since libsixel is using meson already, it would make sense IMHO to use mesonpep517. That creates a PyPI-uploadable package that builds using meson. It may be possible to have it bundle the libsixel shared library rather than requiring it to be installed by hand without `pip install` integration... That would be easier if it were a C binding rather than pure python ctypes. In that case you'd just configure mesonpep517 to build this module using `-Ddefault_library=static` and the resulting python module would be fully standalone.

**Comment by WSLUser** [comment]
Poetry handles dependency management far superior to other solutions like pip. Python packages built with poetry can still be uploaded to PyPi and users can still use pip if they wish to install. I would prefer python bindings remain pure python.

**Pull request merged**
